saleor/checkout/tests/test_calculations.py

```diff
@@ -0,0 +1,189 @@
+from ..fetch import CheckoutLineInfo, fetch_checkout_info, fetch_checkout_lines
+
+
+@pytest.fixture
+def checkout_with_items_lines(checkout_with_items):
+    return checkout_with_items.lines.all()
+
+
+@pytest.fixture
+def tax_data(checkout_with_items, checkout_with_items_lines):
+    checkout = checkout_with_items
+    net = Decimal("10.000")
+    gross = Decimal("12.300")
+    lines = checkout_with_items_lines
+    return TaxData(
+        currency=checkout.currency,
+        shipping_price_net_amount=checkout.shipping_price.net.amount + net,
+        shipping_price_gross_amount=checkout.shipping_price.gross.amount + gross,
+        subtotal_net_amount=checkout.subtotal.net.amount + net,
+        subtotal_gross_amount=checkout.subtotal.gross.amount + gross,
+        total_net_amount=checkout.shipping_price.net.amount + net,
+        total_gross_amount=checkout.shipping_price.gross.amount + gross,
+        lines=[
+            TaxLineData(
+                id=line.id,
+                currency=checkout.currency,
+                unit_net_amount=line.unit_price.net.amount + net,
+                unit_gross_amount=line.unit_price.gross.amount + gross,
+                total_net_amount=line.total_price.net.amount + net,
+                total_gross_amount=line.total_price.gross.amount + gross,
+            )
+            for line in lines
+        ],
+    )
+
+
+def test_apply_tax_data(checkout_with_items, checkout_with_items_lines, tax_data):
+    # given
+    checkout = checkout_with_items
+    lines = checkout_with_items_lines
+
+    def qp(amount):
+        return quantize_price(amount, checkout.currency)
+
+    # when
+    _apply_tax_data(
+        checkout,
+        [
+            Mock(spec=CheckoutLineInfo, line=line, variant=line.variant)
+            for line in lines
+        ],
+        tax_data,
+    )
+
+    # then
+    assert str(checkout.total.net.amount) == str(qp(tax_data.total_net_amount))
+    assert str(checkout.total.gross.amount) == str(qp(tax_data.total_gross_amount))
+
+    assert str(checkout.subtotal.net.amount) == str(qp(tax_data.subtotal_net_amount))
+    assert str(checkout.subtotal.gross.amount) == str(
+        qp(tax_data.subtotal_gross_amount)
+    )
+
+    assert str(checkout.shipping_price.net.amount) == str(
+        qp(tax_data.shipping_price_net_amount)
+    )
+    assert str(checkout.shipping_price.gross.amount) == str(
+        qp(tax_data.shipping_price_gross_amount)
+    )
+
+    for line, tax_line in zip(lines, tax_data.lines):
+        assert str(line.unit_price.net.amount) == str(qp(tax_line.unit_net_amount))
+        assert str(line.unit_price.gross.amount) == str(qp(tax_line.unit_gross_amount))
+
+        assert str(line.total_price.net.amount) == str(qp(tax_line.total_net_amount))
+        assert str(line.total_price.gross.amount) == str(
+            qp(tax_line.total_gross_amount)
+        )
+
+
+@pytest.fixture
+def manager():
+    return get_plugins_manager()
```

Comment:
Don't we have this fixture already in project?

Reply:
I couldn't find one. Every test I've seen used
```python3
#given
manager = get_plugins_manager()
```